include/posix/fs.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __POSIX_FILE_H__
+#define __POSIX_FILE_H__
+
+#include <stdio.h>
+#include <fs.h>
+#include "sys/stat.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct fs_dir_t DIR_t;
+typedef unsigned int mode_t;
+
+typedef struct {
+	unsigned int d_ino;
+	char *d_name;
+} dirent;
+
+/* File related operations */
+extern int open(const char *name, int flags);
+extern int close(int file);
+extern int write(int file, char *buffer, unsigned int count);
+extern int read(int file, char *buffer, unsigned int count);
```

Comment:
POSIX defines return value to be of type ssize_t. Similarly with params and other functions.